### PR TITLE
fix(evmengine): handle engine errors

### PR DESCRIPTION
**Pull request by 0xHansLee**
This PR is the commit from Omni to handle errors of engineAPI (https://github.com/omni-network/omni/pull/2469).

issue: none

### CODE REVIEW

File: client/x/evmengine/keeper/abci_internal_test.go [hunk 1]
```diff
@@ -407,31 +407,33 @@ func TestKeeper_PostFinalize(t *testing.T) {
...
-						PayloadStatus: eengine.PayloadStatusV1{
-							Status:          "unknown status",
-							LatestValidHash: nil,
-							ValidationError: nil,
-						},
-						PayloadID: &payloadID,
-					}, nil
+		/*
+			{
+				name: "fail: unknown status from EL",
```
Comment (0xHansLee) [inline client/x/evmengine/keeper/abci_internal_test.go:412]: Commented this test out in this [commit](https://github.com/piplabs/story/pull/372/commits/b9e0b91f8345dcce7a26ad34611e7a11804c2dd6) as unknown status of payload is not handled anymore.

File: client/app/prouter.go [hunk 2]
```diff
@@ -29,6 +35,11 @@ func makeProcessProposalRouter(app *App) *baseapp.MsgServiceRouter {
 // It also updates some external state.
 func makeProcessProposalHandler(router *baseapp.MsgServiceRouter, txConfig client.TxConfig) sdk.ProcessProposalHandler {
 	return func(ctx sdk.Context, req *abci.RequestProcessProposal) (*abci.ResponseProcessProposal, error) {
+		// Only allow 10s to process a proposal. Reject proposal otherwise.
+		timeoutCtx, timeoutCancel := context.WithTimeout(ctx.Context(), processTimeout)
```
Comment (edisonz0718) [inline client/app/prouter.go:39]: Here in the comment it says allowing 10s to process, but processTimeout is set to 1 Minute. Do I miss anything here?

File: client/x/evmengine/keeper/abci_internal_test.go (hunk 1 above)
Comment (jdubpark) [reply client/x/evmengine/keeper/abci_internal_test.go:412]: Should we just delete it?

**Review by jdubpark — APPROVED** [review]

File: client/x/evmengine/keeper/abci_internal_test.go (hunk 1 above)
Comment (0xHansLee) [reply client/x/evmengine/keeper/abci_internal_test.go:412]: I left it just in case, but I think we can delete it.

Comment (0xHansLee) [reply client/x/evmengine/keeper/abci_internal_test.go]: deleted in [d37a1e0](https://github.com/piplabs/story/pull/372/commits/d37a1e02eb93b66ca251ba38a5c6c99df82d544a).

**Pull request merged**